redash/query_runner/google_spreadsheets.py

```diff
@@ -88,14 +90,27 @@ def __init__(self, worksheet_num, worksheet_count):
         super(WorksheetNotFoundError, self).__init__(message)
 
 
+class WorksheetNotFoundByTitleError(Exception):
+    def __init__(self, worksheet_title):
+        message = "Worksheet title '{}' not found.".format(worksheet_title)
+        super(WorksheetNotFoundByTitleError, self).__init__(message)
+
+
 def parse_query(query):
     values = query.split("|")
     key = values[0]  # key of the spreadsheet
-    worksheet_num = (
-        0 if len(values) != 2 else int(values[1])
-    )  # if spreadsheet contains more than one worksheet - this is the number of it
+    worksheet_num_or_title = 0  # A default value for when a number of inputs is invalid
+    if len(values) == 2:
+        s = values[1].strip()
+        if len(s) > 0:
+            if re.match(r"^\"(.*?)\"$", s):
+                # A string quoted by " means a title of worksheet
+                worksheet_num_or_title = s[1:-1]
+            else:
+                # if spreadsheet contains more than one worksheet - this is the number of it
+                worksheet_num_or_title = int(s)
 
-    return key, worksheet_num
+    return key, worksheet_num_or_title
```

Comment:
I'm not a fan of mixed-type variables that can be one type or another. It makes it difficult to track what the contents of a variable might be like, and means we need to add a lot of "plumbing" type checking code, e.g. L135-140.